python/tests/lis/test_curves.py

```diff
@@ -42,3 +44,544 @@ def test_read_curves():
+
+        fmt = core.dfs_formatstring(dfs)
+        assert fmt == 'B8OI1D2F'
+
+        curves = lis.curves(f, dfs)
+        assert curves['BYTE'] == [89]
+        assert curves['I8  '] == [-128]
+        assert curves['I16 '] == [153]
+        assert curves['I32 '] == [-153]
+        assert curves['F16 '] == [1.0]
+        assert curves['F32 '] == [-1.0]
+        assert curves['F32L'] == [-0.25]
+        assert curves['F32F'] == [153.25]
+
+@pytest.mark.xfail(strict=True, reason="strings would be supported soon")
+def test_fdata_repcodes_string(tmpdir, merge_lis_prs):
+    fpath = os.path.join(str(tmpdir), 'fdata-repcodes-string.lis')
+
+    content = headers + [
+        'data/lis/records/curves/dfsr-repcodes-string.lis.part',
+        'data/lis/records/curves/fdata-repcodes-string.lis.part',
+    ] + trailers
+
+    merge_lis_prs(fpath, content)
+
+    with lis.load(fpath) as (f,):
+        dfs = f.data_format_specs()[0]
+
+        fmt = core.dfs_formatstring(dfs)
+        assert fmt == 'A'
+
+        curves = lis.curves(f, dfs)
+        assert curves['STR '] == "Now this is a string of size 32 "
+
+def test_fdata_repcodes_mask(tmpdir, merge_lis_prs):
+    fpath = os.path.join(str(tmpdir), 'fdata-repcodes-mask.lis')
+
+    content = headers + [
+        'data/lis/records/curves/dfsr-repcodes-mask.lis.part',
+        'data/lis/records/curves/fdata-repcodes-mask.lis.part',
+    ] + trailers
+
+    merge_lis_prs(fpath, content)
+
+    with lis.load(fpath) as (f,):
+        dfs = f.data_format_specs()[0]
+
+        with pytest.raises(NotImplementedError):
+            fmt = core.dfs_formatstring(dfs)
+            assert fmt == 'M'
+
+        with pytest.raises(NotImplementedError):
+            curves = lis.curves(f, dfs)
+            assert curves['MASK'] == bytearray([0xFF, 0xFF, 0xFF, 0xFF])
+
+def test_fdata_repcodes_invalid(tmpdir, merge_lis_prs):
+    fpath = os.path.join(str(tmpdir), 'fdata-repcodes-invalid.lis')
+
+    content = headers + [
+        'data/lis/records/curves/dfsr-repcodes-invalid.lis.part',
+    ] + trailers
+
+    merge_lis_prs(fpath, content)
+
+    with lis.load(fpath) as (f,):
+        dfs = f.data_format_specs()[0]
+
+        with pytest.raises(RuntimeError) as exc:
+            _ = lis.curves(f, dfs)
+        assert "Invalid repcode (170) in channel (WTF )" in str(exc.value)
+
+
+# DEPTH:
+# 1. placeholders. currently expected behavior may be changed on implementation
+# 2. more tests needed if depth can be negative
+# 3. might need test mode = 0, direction flag inconsistent with actual direction
+# if we decide to implement any checks
+# 4. Issues with inconsistent spacing and units: if depth units (14) != frame
+# spacing units (9), we might have problems with defining inconsistency
+# 5. Some information can be check or deduced if certain entries are missing.
+# Suggestion is to fail on all wrong or missing important information cases
+# until we get the real file with these situations.
```

Comment:
Totally agree! Let's not implement _any_ value-deduction before we actually see a need for it. Great that you have comments everywhere that this would be a possibility 👍